client/my-sites/plugins/plugin-upload/index.jsx

```diff
@@ -0,0 +1,146 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import page from 'page';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import Main from 'components/main';
+import HeaderCake from 'components/header-cake';
+import Card from 'components/card';
+import ProgressBar from 'components/progress-bar';
+import DropZone from 'components/drop-zone';
+import FilePicker from 'components/file-picker';
+import { uploadPlugin, clearPluginUpload } from 'state/plugins/upload/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import {
+	getPluginUploadError,
+	getPluginUploadProgress,
+	getUploadedPluginId,
+	isPluginUploadComplete,
+	isPluginUploadInProgress,
+} from 'state/selectors';
+import notices from 'notices';
+import {
+	MAX_UPLOADED_THEME_SIZE
+} from 'lib/automated-transfer/constants';
+
+class PluginUpload extends React.Component {
+	back = () => {
+		page.back();
+	}
+
+	renderUploadCard() {
+		const { inProgress, complete } = this.props;
+		return (
+			<Card>
+				{ ! inProgress && ! complete && this.renderDropZone() }
+				{ inProgress && this.renderProgressBar() }
+			</Card>
+		);
+	}
+
+	onFileSelect = ( files ) => {
+		const { translate, siteId } = this.props;
+		const errorMessage = translate( 'Please drop a single zip file' );
+
+		if ( files.length !== 1 ) {
+			notices.error( errorMessage );
+			return;
+		}
+
+		// DropZone supplies an array, FilePicker supplies a FileList
+		const file = files[ 0 ] || files.item( 0 );
+
+		// TODO: plugin-specific constant
+		if ( file.size > MAX_UPLOADED_THEME_SIZE ) {
+			notices.error(
+				translate( 'Plugin zip is too large. Please upload a plugin under 50 MB.' )
+			);
+
+			return;
+		}
+
+		this.props.uploadPlugin( siteId, file );
+	}
+
+	renderDropZone() {
+		const { translate } = this.props;
+		const dropText = translate(
+			'Drop files or click here to upload'
+		);
+		const uploadInstructionsText = translate(
+			'Only single .zip files are accepted.'
+		);
+
+		return (
+			<div className="plugin-upload__dropzone">
+				<DropZone onFilesDrop={ this.onFileSelect } />
+				<FilePicker accept="application/zip" onPick={ this.onFileSelect } >
+					<Gridicon
+						className="plugin-upload__dropzone-icon"
+						icon="cloud-upload"
+						size={ 48 } />
+					{ dropText }
+					<span className="plugin-upload__dropzone-instructions">{ uploadInstructionsText }</span>
+				</FilePicker>
+			</div>
+		);
+	}
+
+	renderProgressBar() {
+		const { translate, progress, installing } = this.props;
+
+		const uploadingMessage = translate( 'Uploading your plugin' );
+		const installingMessage = translate( 'Installing your plugin' );
```

Comment:
Hi! I've found a possible matching string that has already been translated **53** times:
`translate( 'Installing the plugin&#8230;' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).